app/models/split.rb

```diff
@@ -82,8 +119,34 @@ def create_decision!(params = {})
     build_decision(params).tap(&:save!)
   end
 
+  def registry
+    if try(:feature_incomplete?)
+      knock_out_weightings(super)
+    else
+      super
+    end
+  end
+
   private
 
+  def knock_out_weightings(registry_hash, to: "false")
+    target_variant = to.to_s
+    knock_out_weightings_if_possible(registry_hash, to: target_variant) || log_knockout_error(target_variant) && registry_hash
+  end
+
+  def knock_out_weightings_if_possible(registry_hash, to:)
+    found_key = false
```

Comment:
What if we expressed this as `found_key = registry_hash.key?(to)` and short-circuit out of the whole thing?

Reply:
(is the concern that we need the `to_s` on all the keys? Can we `stringify_keys` first? Also, if they're coming straight from the DB, won't they always be strings?)

Reply:
I get that this isn’t the prettiest on the inside. Lots of options, I think. I just suspect this code will never change after this. I’ve already got the function I need to knock out non-feature gates for remote kill scenarios.

Reply:
Re always be strings, the object has setters, I’m just shooting for the principle of least surprise if somebody makes a weird usage decision later. I can’t make them not set the value of registry, that’s not what readonly means iirc; just resists saves.